### PR TITLE
DOCSUP-1585: Edit and sync geo functions

### PR DESCRIPTION
**Pull request by olgarev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation

Changelog entry

Translated into English: h3IsValid, h3EdgeAngle, h3EdgeLengthM, h3kRing.
Translated into Russian: geohashesInBox.
Functions organized into 2 subsections: Geohash and H3 Indexes.

### CODE REVIEW

File: docs/en/sql-reference/functions/geo.md [hunk 1]
```diff
@@ -135,9 +135,17 @@ SELECT pointInPolygon((3., 3.), [(6, 0), (8, 4), (5, 8), (0, 2)]) AS res
...
 
-## geohashEncode {#geohashencode}
+## Functions for Working with Geohash {#geohash}
 
-Encodes latitude and longitude as a geohash-string, please see (http://geohash.org/, https://en.wikipedia.org/wiki/Geohash).
+Geohash is the geocode system, which subdivides Earth’s surface into buckets of grid shape and encodes each cell into a short string of letters and digits. It is a hierarchical data structure, so the longer is the geohash string, the more precise is the geographic location.  
+
+The description of the Geohash system is available in the [Wikipedia](https://en.wikipedia.org/wiki/Geohash). 
+
+If you need to convert geographic coordinates to geohash strings, you can use [geohash.org](http://geohash.org/).
```
Comment (blinkov) [inline docs/en/sql-reference/functions/geo.md:144]:
```suggestion
[Geohash](https://en.wikipedia.org/wiki/Geohash) is the geocode system, which subdivides Earth’s surface into buckets of grid shape and encodes each cell into a short string of letters and digits. It is a hierarchical data structure, so the longer is the geohash string, the more precise is the geographic location. 

If you need to manually convert geographic coordinates to geohash strings, you can use [geohash.org](http://geohash.org/).
```

File: docs/ru/sql-reference/functions/geo.md [hunk 2]
```diff
@@ -128,9 +128,17 @@ SELECT pointInPolygon((3., 3.), [(6, 0), (8, 4), (5, 8), (0, 2)]) AS res
...
 
-## geohashEncode {#geohashencode}
+## Функции для работы с системой Geohash {#geohash}
 
-Кодирует широту и долготу в строку geohash, смотрите http://geohash.org/, https://en.wikipedia.org/wiki/Geohash.
+Geohash — это система геокодирования, которая делит поверхность Земли на участки в виде "решетки", и каждую ячейку решетки кодирует в виде строки из букв и цифр. Система поддерживает иерархию (вложенность) ячеек, поэтому чем точнее определена геопозиция, тем длиннее строка с кодом соответствующей ячейки.  
+
+Описание системы Geohash можно посмотреть в [Википедии](https://en.wikipedia.org/wiki/Geohash). 
+
+Для преобразования географических координат в строку geohash можно использовать сайт [geohash.org](http://geohash.org/).
```
Comment (blinkov) [inline docs/ru/sql-reference/functions/geo.md:137]:
```suggestion
[Geohash](https://en.wikipedia.org/wiki/Geohash) — это система геокодирования, которая делит поверхность Земли на участки в виде "решетки", и каждую ячейку решетки кодирует в виде строки из букв и цифр. Система поддерживает иерархию (вложенность) ячеек, поэтому чем точнее определена геопозиция, тем длиннее строка с кодом соответствующей ячейки. 

Для ручного преобразования географических координат в строку geohash можно использовать сайт [geohash.org](http://geohash.org/).
```

File: docs/en/sql-reference/functions/geo.md [hunk 3]
```diff
@@ -135,9 +135,17 @@ SELECT pointInPolygon((3., 3.), [(6, 0), (8, 4), (5, 8), (0, 2)]) AS res
 └─────┘
 ```
 
-## geohashEncode {#geohashencode}
+## Functions for Working with Geohash {#geohash}
```
Comment (blinkov) [inline docs/en/sql-reference/functions/geo.md:138]: we could split all this into:
```
geo/index.md
geo/geohash.md
geo/h3.md
```

File: docs/en/sql-reference/functions/geo.md [hunk 4]
```diff
@@ -189,13 +197,205 @@ SELECT geohashDecode('ezs42') AS res
...
+**Parameters**
+
+-   `longitude_min` — Minimum longitude. Range: `[-180°, 180°]`. Type: [Float](../../sql-reference/data-types/float.md).
+-   `latitude_min` — Minimum latitude. Range: `[-90°, 90°]`. Type: [Float](../../sql-reference/data-types/float.md).
+-   `longitude_max` — Maximum longitude. Range: `[-180°, 180°]`. Type: [Float](../../sql-reference/data-types/float.md).
+-   `latitude_max` — Maximum latitude. Range: `[-90°, 90°]`. Type: [Float](../../sql-reference/data-types/float.md).
+-   `precision` — Geohash precision. Range: `[1, 12]`. Type: [UInt8](../../sql-reference/data-types/int-uint.md).
+
+!!! info "Note"
+    All coordinate parameters should be of the same type: either `Float32` or `Float64`.
```
Comment (BayoNet) [inline docs/en/sql-reference/functions/geo.md:219]:
```suggestion
    All coordinate parameters must be of the same type: either `Float32` or `Float64`.
```

File: docs/en/sql-reference/functions/geo.md [hunk 5]
```diff
@@ -189,13 +197,205 @@ SELECT geohashDecode('ezs42') AS res
...
+
+**Returned values**
+
+-   Array of precision-long strings of geohash-boxes covering provided area, you should not rely on order of items.
+-   \[\] - Empty array if *min* values of *latitude* and *longitude* aren’t less than corresponding *max* values.
```
Comment (BayoNet) [inline docs/en/sql-reference/functions/geo.md:224]:
```suggestion
-   `[]` - Empty array if minimum latitude and longitude values aren’t less than corresponding maximum values.
```

File: docs/en/sql-reference/functions/geo.md [hunk 6]
```diff
@@ -189,13 +197,205 @@ SELECT geohashDecode('ezs42') AS res
...
+
+Type: [Array](../../sql-reference/data-types/array.md)([String](../../sql-reference/data-types/string.md)).
+
+!!! info "Note"
+    Function will throw an exception if resulting array is over 10’000’000 items long.
```
Comment (BayoNet) [inline docs/en/sql-reference/functions/geo.md:229]:
```suggestion
    Function throws an exception if resulting array is over 10’000’000 items long.
```

File: docs/en/sql-reference/functions/geo.md [hunk 7]
```diff
@@ -189,13 +197,205 @@ SELECT geohashDecode('ezs42') AS res
...
+
+[H3](https://eng.uber.com/h3/) is a geographical indexing system where Earth’s surface divided into a grid of even hexagonal cells. This system is hierarchical, i. e. each hexagon on the top level ("parent") can be splitted into seven even but smaller ones ("children"), and so on. 
+
+The level of the hierarchy is called `resolution` and can receive a value from `0` till `15`, where `0` is the `base` level with the largest and coarsest cells. 
+
+A latitude and longitude pair can be transformed to a 64-bit H3 index, identifying a grid cell.
+
+The H3 index is used primarily for bucketing locations and other geospatial manipulations.
+
+The full description of the H3 systen is available at [the Uber Engeneering site](https://eng.uber.com/h3/).
```
Comment (BayoNet) [inline docs/en/sql-reference/functions/geo.md:256]:
```suggestion
The full description of the H3 system is available at [the Uber Engeneering site](https://eng.uber.com/h3/). 
```

File: docs/en/sql-reference/functions/geo.md (hunk 3 above)
Comment (BayoNet) [reply docs/en/sql-reference/functions/geo.md:138]: We will do it in this PR.

**Pull request merged**